src/Api/Test/ClientTest.php

```diff
@@ -0,0 +1,572 @@
+            'hydra:description' => 'The client "0" does not exist.',
+        ]);
+    }
+
+    public function testGetCollection(): void
+    {
+        $httpClient = $this->authenticate();
+        $timestamp = $this->getTimestamp();
+        $httpClient->request('POST', '/api/clients', [
+            'json' => [
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client' . $timestamp,
+                'owner' => 1,
+                'quota' => - 1
+            ]
+        ]);
+        $response = $httpClient->request('GET', '/api/clients');
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+        '@context' => '/api/contexts/Client',
+        '@type' => 'hydra:Collection',
+        '@id' => '/api/clients',
+        ]);
+    }
+
+    public function testGetCollectionUnauthenticated(): void
+    {
+        $response = static::createClient()->request('GET', '/api/clients');
+        $this->assertResponseStatusCodeSame(401);
+    }
+
+    public function testGetInvalidItem(): void
+    {
+        $httpClient = $this->authenticate();
+        $timestamp = $this->getTimestamp();
+        $httpClient->request('POST', '/api/clients', [
+            'json' => [
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client' . $timestamp,
+                'owner' => 1,
+                'quota' => - 1
+            ]
+        ]);
+        $response = $httpClient->request('GET', '/api/clients/' . $timestamp);
+        $this->assertResponseStatusCodeSame(404);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+    }
+
+    public function testGetItem(): void
+    {
+        $httpClient = $this->authenticate();
+        $timestamp = $this->getTimestamp();
+        $httpClient->request('POST', '/api/clients', [
+            'json' => [
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client' . $timestamp,
+                'owner' => 1,
+                'quota' => - 1
+            ]
+        ]);
+        $iri = $this->findIriBy(Client::class, [
+            'name' => 'client' . $timestamp
+        ]);
+        $response = $httpClient->request('GET', $iri);
+
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+            'isActive' => true,
+            'maxParallelJobs' => 1,
+            'name' => 'client' . $timestamp,
+            'owner' => 1,
+            'postScripts' => [],
+            'preScripts' => [],
+            'quota' => '-1',
+            'rsyncLongArgs' => null,
+            'rsyncShortArgs' => null,
+            'sshArgs' => null,
+            'url' => ""
+        ]);
```

Comment:
This test might be more interesting if you get 2 clients, one with the minimun (like the one you already have) and one with all parameters set.
This way it might also catch serialization related issues.